File: Master/Web Services and Middleware Technologies/individual-project-lauradeac-main/backend/src/main/java/com/backend/domain/dto/SongDTO.java
package com.backend.domain.dto;

import com.backend.domain.entity.Genre;
import com.backend.domain.entity.Playlist;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class SongDTO {

    private String songId;

    private String title;

    private String artist;

    private Integer duration;

    private String genre;

    private String playlistId;
}

File: Master/Web Services and Middleware Technologies/individual-project-lauradeac-main/frontend/src/app/model/songDTO.ts
export class SongDTO {
  songId?: string | undefined;
  title!: string;
  artist!: string;
  duration!: number;
  genre!: string;
  playlistId!: string;
  isEditable?: boolean;

  constructor(
    title: string,
    artist: string,
    duration: number,
    genre: string,
    playlistId: string,
    songId?: string,
    isEditable?: boolean
  ) {
    this.title = title;
    this.artist = artist;
    this.duration = duration;
    this.genre = genre;
    this.playlistId = playlistId;
    this.songId = songId;
    this.isEditable = false;
  }
}
